src/orange/construct/OrangeInputIO.json.hh

```diff
@@ -1,21 +1,24 @@
 //----------------------------------*-C++-*----------------------------------//
-// Copyright 2021-2022 UT-Battelle, LLC, and other Celeritas developers.
+// Copyright 2022 UT-Battelle, LLC, and other Celeritas developers.
 // See the top-level COPYRIGHT file for details.
 // SPDX-License-Identifier: (Apache-2.0 OR MIT)
 //---------------------------------------------------------------------------//
-//! \file orange/construct/SurfaceInputIO.json.hh
+//! \file orange/construct/OrangeInputIO.json.hh
 //---------------------------------------------------------------------------//
 #pragma once
 
 #include <nlohmann/json.hpp>
 
-#include "SurfaceInput.hh"
+#include "OrangeInput.hh"
 
 namespace celeritas
 {
 //---------------------------------------------------------------------------//
 
 void from_json(const nlohmann::json& j, SurfaceInput& value);
+void from_json(const nlohmann::json& j, VolumeInput& value);
+void from_json(const nlohmann::json& j, UnitInput& value);
+void from_json(const nlohmann::json& j, OrangeInput& value);
```

Comment:
Ok - I'm having a really hard time following where these are used (this file is included in `OrangeParams.cc` but can't find where these are being called).  So - this logic is difficult for me to understand.  I would have expected there to be a single `from_json(j, OrangeInput& value)` that called the units, which called the volumes, which called the surfaces, etc.  Since that mirroros the `OrangeInput.hh` structs.

This would probably make sense if I saw where they were being called, but I didn't see that (probably me being obtuse here).

Reply:
They're called under the hood by the `j.get_to` or `j.get<OrangeInput>()` functions... it's part of the nljson API.